### PR TITLE
[Core] Some small fixes and improvements

### PR DESCRIPTION
**Pull request by kfstorm**
## Why are these changes needed?

There are some small fixes and improvements we made in Ant internal repo but forgot to upstream to the community.

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: src/ray/core_worker/core_worker.cc [hunk 1]
```diff
@@ -308,6 +308,8 @@ CoreWorker::CoreWorker(const CoreWorkerOptions &options, const WorkerID &worker_
       task_execution_service_work_(task_execution_service_),
       resource_ids_(new ResourceMappingType()),
       grpc_service_(io_service_, *this) {
+  RAY_LOG(INFO) << "Constructing CoreWorker, worker_id: " << worker_id;
```
Comment (rkooo567) [inline src/ray/core_worker/core_worker.cc:311]: Can we actually add a pid here too?

File: src/ray/core_worker/store_provider/plasma_store_provider.cc [hunk 2]
```diff
@@ -79,9 +79,7 @@ CoreWorkerPlasmaStoreProvider::~CoreWorkerPlasmaStoreProvider() {
 
 Status CoreWorkerPlasmaStoreProvider::SetClientOptions(std::string name,
                                                        int64_t limit_bytes) {
-  std::lock_guard<std::mutex> guard(store_client_mutex_);
-  RAY_RETURN_NOT_OK(store_client_.SetClientOptions(name, limit_bytes));
-  return Status::OK();
+  return store_client_.SetClientOptions(name, limit_bytes);
```
Comment (rkooo567) [inline src/ray/core_worker/store_provider/plasma_store_provider.cc:82]: Don't we still need a lock?

**Review by rkooo567 — COMMENTED** [review]
Why do we remove locks from the plasma store provider?

File: src/ray/core_worker/store_provider/plasma_store_provider.cc (hunk 2 above)
Comment (suquark) [reply src/ray/core_worker/store_provider/plasma_store_provider.cc:82]: I think we already have `std::lock_guard<std::recursive_mutex> guard(client_mutex_);` in all store client function calls.

File: src/ray/core_worker/core_worker.cc (hunk 1 above)
Comment (kfstorm) [reply src/ray/core_worker/core_worker.cc:311]: The pid is already logged in `CoreWorkerProcess::CoreWorkerProcess`.

File: src/ray/core_worker/store_provider/plasma_store_provider.cc (hunk 2 above)
Comment (rkooo567) [reply src/ray/core_worker/store_provider/plasma_store_provider.cc:82]: Sounds good. Thanks for the confirmation

**Review by rkooo567 — APPROVED** [review]

**Pull request merged**